src/devtools/ProfilingCache.js

```diff
@@ -365,32 +362,42 @@ export default class ProfilingCache {
     }
   };
 
-  onProfileSummary = ({
-    commitDurations,
-    commitTimes,
-    initialTreeBaseDurations,
-    interactionCount,
-    rootID,
-  }: ProfilingSummaryBackend) => {
+  onProfileSummary = (profilingSummaryBackend: ProfilingSummaryBackend) => {
+    const { rootID } = profilingSummaryBackend;
     const resolve = this._pendingProfileSummaryMap.get(rootID);
     if (resolve != null) {
       this._pendingProfileSummaryMap.delete(rootID);
 
-      const initialTreeBaseDurationsMap = new Map();
-      for (let i = 0; i < initialTreeBaseDurations.length; i += 2) {
-        initialTreeBaseDurationsMap.set(
-          initialTreeBaseDurations[i],
-          initialTreeBaseDurations[i + 1]
-        );
-      }
-
-      resolve({
-        rootID,
-        commitDurations,
-        commitTimes,
-        initialTreeBaseDurations: initialTreeBaseDurationsMap,
-        interactionCount,
-      });
+      resolve(
+        convertProfilingSummaryFromBackendToFrontend(profilingSummaryBackend)
+      );
     }
   };
 }
+
+function convertProfilingSummaryFromBackendToFrontend(
```

Comment:
Why split this out? It's only used/called in one place.

Reply:
> Why split this out? It's only used/called in one place.

Make the intent clear. This is a separate (testable btw) piece of logic. We're splitting into functions to make the code more readable and manageable, not just for DRY.

Reply:
I don't think splitting things into helper functions _necessarily_ improves readability or maintainability. (In this case, I don't think it does.) And it seems sufficient to test this with a more end-to-end test (as part of testing the whole import functionality) rather than in a one-off test.